src/utils/middleware/csrf.middleware.js

```diff
@@ -0,0 +1,79 @@
+const csrf = require('csrf')
+
+/**
+ * @param logger {Logger}
+ * @param sessionKey {string}
+ * @param secretKey {string}
+ * @param tokenKey {string}
+ */
+const configureCsrfMiddleware = (logger, sessionKey, secretKey = 'csrfSecret', tokenKey = 'csrfToken') => {
+  logger.debug('--- CSRF middleware configuration ---')
+  logger.debug(`Secret is set at req.[${sessionKey}][${secretKey}]`)
+  logger.debug(`Token is checked at req.body|query[${tokenKey}]`)
+  logger.debug('-------------------------------------')
+  /**
+   * @param csrfToken {string}
+   * @param csrfSecret {string}
+   * @param req {e.Request}
+   */
+  const csrfValid = (csrfToken, csrfSecret, req) => {
```

Comment:
it would be sufficient to only take the HTTP method as a parameter here, rather than the entire `req` object